Providers/nxOMSAutomationWorker/automationworker/worker/jrdsclient.py

```diff
@@ -78,10 +87,85 @@ def get_sandbox_actions(self):
+                #tracer.log_worker_debug("sandbox_response")
+                #tracer.log_worker_debug(sandbox_response)
+                #print response.headers
+                tracer.log_worker_debug("jrds107")
+
+                #Initiate certificate rotation if Headers is set True
+                tracer.log_worker_debug(workercertificaterotation.get_certificate_rotation_header_value())
+            
+                if(eval(workercertificaterotation.get_certificate_rotation_header_value())):
+                    tracer.log_worker_debug("Initiating certificate Rotation of Hybrid Worker")
+                    workercertificaterotation.set_certificate_rotation_header_value(DISABLE_CERT_ROTATION)  
+                    tracer.log_worker_debug("jrds114")
+                    tracer.log_worker_debug(workercertificaterotation.get_certificate_rotation_header_value())
+                    self.worker_certificate_rotation()
+            except Exception:
+                pass
+
             return response.deserialized_data["value"]
 
         raise Exception("Unable to get sandbox actions. [status=" + str(response.status_code) + "]")
 
+
+    def worker_certificate_rotation(self):
+        """ Rotate worker certificate. 
+        Steps includes creating new certificate and after Server returns 200, replace the old certificate with newly generated certificate.
+
+        """
+        import tracer
+        try:
+            temp_certificate_path, temp_key_path = workercertificaterotation.generate_cert_rotation_self_signed_certificate()
+            issuer, subject, thumbprint, not_before, not_after = linuxutil.get_cert_info(temp_certificate_path)
+
+            payload = {'Thumbprint': thumbprint,
+               'Issuer': issuer,
+               'Subject': subject,
+               'NotBefore': not_before,
+               'NotAfter': not_after}
+
+            headers = {"Content-Type": "application/json"}
+
+            url = self.base_uri + "/automationAccounts/" + self.account_id + \
+              "/hybridCertificateRotation?api-version=" + self.protocol_version
+            response = self.issue_request(lambda u: self.httpClient.post(u, headers=headers, data=payload), url)
+
+            if response.status_code == 200:
+                import tracer
+                tracer.log_worker_debug("Worker certificate successfully got rotated on the server side.")
+                workercertificaterotation.replace_self_signed_certificate_and_key(temp_certificate_path, temp_key_path, thumbprint)
+                tracer.log_worker_debug("replace_self_signed_certificate_and_key done")
+                return
+            else:
+                import tracer
+                tracer.log_worker_debug(response)
+                tracer.log_worker_debug(dir(response))
+                tracer.log_worker_debug(response.status_code)
+
+            workercertificaterotation.clean_up_certificate_and_key(temp_certificate_path, temp_key_path)
```

Comment:
should this be moved to finally block ?
How will we cleanup certs if there is an exception ?

Reply:
done, I was cleaning it up in both cases but finally block makes more sense